android/KMAPro/kMAPro/src/main/java/com/tavultesoft/kmapro/MainActivity.java

```diff
@@ -589,11 +684,35 @@ public void onKeyboardDownloadFinished(HashMap<String, String> keyboardInfo, int
           KMManager.setKeyboard(packageID, keyboardID, languageID, keyboardName, languageName, kFont, kOskFont);
         }
       }
+    } else if (result == 0) {
+      Toast.makeText(this, "Keyboard already exists", Toast.LENGTH_SHORT).show();
```

Comment:
Can we give a clearer error message -- e.g. `"Keyboard 'keyboardID' already exists"`? What are the circumstances in which this would happen? Would we ever run into a problem for a user wanting to upgrade an ad-hoc keyboard that is already installed? (No version numbers in filenames after all)

Reply:
Oops, `result == 0` was from an earlier version of this branch that juggled keyboard upgrades/downgrades. I need to remove this path since it's no longer applicable.